### PR TITLE
feat: adding ability to merge PRs that have finished required status checks

### PR DESCRIPTION
**Pull request by varcheesy123**
https://developer.github.com/v4/enum/mergestatestatus/
Github gives mergeable PRs an `unstable` MergeStateStatus when the required checks have passed but CI is still running.  MergeStateStatus is only set to `clean` when CI has completely finished
This pr simply adds a flag that decides whether or not Github-Mergerine should merge an `unstable` PR or not.

### CODE REVIEW

**Comment by jmccann** [comment]
Also, `unstable` is my understanding from reading people's comments online is actually when all required checks have passed but all other checks are still pending or failed.

So CI could be complete and have failing checks that are not required can cause PR to be `unstable`.

**Comment by varcheesy123** [comment]
`when all required checks have passed but all other checks are still pending or failed`
Isn't that exactly what we want? 
The only other state that Github deems 'mergeable' is `Has Hooks`. Should that also be included here?

File: src/nocks.js [hunk 1]
```diff
@@ -467,6 +468,37 @@ export const none = () => {
...
+    .reply(200, restrictions)
+    .get(uri => uri.includes('/members'))
+    .times(Infinity)
+    .reply(200, [
+      {
+        login: 'SuperUser'
+      }
+    ])
+    .put(uri => uri.includes('/merge'))
+    .reply(404)
```
Comment (AndersDJohnson) [inline src/nocks.js]: @varcheesy123 Why do you want the merge call to 404 here?

**Comment by AndersDJohnson** [comment]
@jmccann Having a whitelist of statuses in the config would expose more options to the users, which has value especially if GitHub changes or adds statuses in the future and we don't update our library or users can't update for other reasons such as backward compatibility. However, such a low level option would also require them to either copy/paste from an example configuration, or at worst to list and understand all the mergeable state statuses - the documentation for which is pretty poor and not very discoverable. I think it's okay to provide a simple boolean in the config that abstracts on top of the known or common mergeable statuses, as @varcheesy123 has implemented. We could later update it to include additional mergeable statuses like `HAS_HOOKS`. Then separately later we could expose the full whitelist configuration option as well. Thoughts?

File: src/config.js [hunk 2]
```diff
@@ -43,6 +44,9 @@ const deleteBranchAfterMerge =
 
 const dry = MERGERINE_DRY === 'true' || config.dry
 
+const mergeableStateCheck =
+  MERGERINE_MERGEABLE_STATE_CHECK === 'false' || config.mergeableStateCheck
```
Comment (AndersDJohnson) [inline src/config.js]: Shouldn't this check for `MERGERINE_MERGEABLE_STATE_CHECK === 'true'`? Was your intent to enable it by default? If so, perhaps it should've been:
```js
MERGERINE_MERGEABLE_STATE_CHECK ?
  MERGERINE_MERGEABLE_STATE_CHECK !== 'false' : 
  config.mergeableStateCheck
```

Comment (AndersDJohnson) [reply src/config.js]: Fixed this, with new unit tests to prove it, in 2320a29b8292484223a4dd13b650389b79121271, published with `1.10.1`.

**Pull request merged**